### PR TITLE
initial accelsim traing branch

### PR DESCRIPTION
**Pull request by gongchen618**
did some refactor, make the trace data flow through, no output

### CODE REVIEW

**Comment by gongchen618** [comment]
😮I think the lint test isn't working, I definitely cannot pass.

**Comment by syifan** [comment]
> 😮I think the lint test isn't working, I definitely cannot pass.

Actually, all the tests passed. So, there is no linting issue.

**Review by syifan — CHANGES_REQUESTED** [review]

File: acceclsim_tracing/example/bfs-rodinia-2.0-ft/kernel-1.traceg
[the remark this answers is not in the document]
Comment (gongchen618) [reply acceclsim_tracing/example/bfs-rodinia-2.0-ft/kernel-1.traceg:1]: yes, lets remove it.

File: acceclsim_tracing/gpu/builder.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+		aluNum  int32
+	}{aluType: aluType, aluNum: num})
+	return g
+}
+
+func (g *GPU) Build() {
+	g.buildDispatcher()
+	g.gpcs = make([]*gpc.GPC, g.meta.gpcNum)
+	for i := 0; i < int(g.meta.gpcNum); i++ {
+		g.gpcs[i] = gpc.NewGPC().WithSMNum(g.meta.smNum).WithSMUnitNum(g.meta.smUnitNum).
```
Comment (syifan) [inline acceclsim_tracing/gpu/builder.go:135]: One line per with-function

File: acceclsim_tracing/trace/convert.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
+package trace
+
+import "github.com/sarchlab/mgpusim/accelsim_tracing/nvidia"
+
+func (tb *threadBlock) generateNVThreadBlock() *nvidia.ThreadBlock {
+	nvtb := &nvidia.ThreadBlock{
+		WarpNum: len(tb.warps),
+	}
+	for _, wp := range tb.warps {
```
Comment (syifan) [inline acceclsim_tracing/trace/convert.go:9]: Empty line before for loop

File: acceclsim_tracing/trace/convert.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+		nvtb.Warps = append(nvtb.Warps, wp.generateNVWarp())
+	}
+	return nvtb
```
Comment (syifan) [inline acceclsim_tracing/trace/convert.go:12]: Empty line before return

**Review by syifan — APPROVED** [review]

**Review by syifan — DISMISSED** [review]

**Review by syifan — DISMISSED** [review]

**Pull request merged**